src/status_im/events.cljs

```diff
@@ -2149,3 +2152,119 @@
  :shake-event
  (fn [cofx _]
    (logging/show-logs-dialog cofx)))
+
+(re-frame/reg-fx
+ ::ens/resolve
+ (fn [[registry name cb]]
+   (ens/get-addr registry name cb)))
+
+(defn- on-resolve [registry custom-domain? username address public-key s]
+  (cond
+    (= (ethereum/normalized-address address) (ethereum/normalized-address s))
+    (resolver/pubkey registry username
+                     (fn [ss]
+                       (if (= ss public-key)
+                         (re-frame/dispatch [:ens/set-state :connected])
+                         (re-frame/dispatch [:ens/set-state :owned]))))
+
+    (and (nil? s) (not custom-domain?)) ;; No address for a stateofus subdomain: it can be registered
+    (re-frame/dispatch [:ens/set-state :registrable])
+
+    :else
+    (re-frame/dispatch [:ens/set-state :unregistrable])))
+
+(defn- assoc-state [db state]
+  (assoc-in db [:ens :state] state))
+
+(defn- assoc-username [db username]
+  (assoc-in db [:ens :username] username))
+
+(defn- valid-custom-domain? [username]
+  (and (ens/is-valid-eth-name? username)
+       (stateofus/lower-case? username)))
+
+(defn- valid-username? [custom-domain? username]
+  (if custom-domain?
+    (valid-custom-domain? username)
+    (stateofus/valid-username? username)))
+
+(handlers/register-handler-fx
+ :ens/set-state
```

Comment:
note that if you use defn/fx to define the functions used in these events you can avoid declaring them here and simply use the attribute map key like this:

```clojure
(fx/defn set-state
  {:events [:ens/set-state]}
  [{:keys [db]} state]
 {:db (assoc-state db state)})
```